lib/puppet/indirector/facts/rest.rb

```diff
@@ -4,6 +4,25 @@
 class Puppet::Node::Facts::Rest < Puppet::Indirector::REST
   desc "Find and save facts about nodes over HTTP via REST."
 
+  def find(request)
```

Comment:
Is there a reason you're not including `return super unless use_http_client?` here?

Reply:
Whoops, not intentional. Let me fix that.

Reply:
Done